qctrlopencontrols/dynamic_decoupling_sequences/predefined.py

```diff
@@ -912,29 +1095,55 @@ def new_x_concatenated_sequence(
 def new_xy_concatenated_sequence(
     duration=1.0, concatenation_order=None, pre_post_rotation=False, **kwargs
 ):
-    """
-    Creates XY-Concatenated Dynamic Decoupling Sequence.
-    Concatenation of base sequence C(\tau/4)XC(\tau/4)YC(\tau/4)XC(\tau/4)Y
+    r"""
+    Creates the :math:`XY`-Concatenated sequence.
 
     Parameters
     ----------
     duration : float, optional
-        defaults to None
-        The total duration of the sequence
+        Defaults to None.
+        The total duration of the sequence :math:`\tau`.
     concatenation_order : int, optional
-        defaults to None
-        The number of concatenation of base sequence
+        Defaults to None.
+        The number of concatenation of base sequence :math:`l`.
     pre_post_rotation : bool, optional
-        If True, a :math:`\\pi.2` rotation is added at the
+        If True, a :math:`X_{\pi/2}` rotation is added at the
         start and end of the sequence.
     kwargs : dict
-        Additional keywords required by
-        qctrlopencontrols.dynamical_decoupling_sequences.DynamicDecouplingSequence
+        Additional keywords required by DynamicDecouplingSequence.
 
     Returns
     -------
-    qctrlopencontrols.dynamical_decoupling_sequences.DynamicDecouplingSequence
-        XY concatenated sequence.
+    DynamicDecouplingSequence
+        The :math:`XY`-concatenated sequence.
+
+    See Also
+    --------
+    new_x_concatenated_sequence
+
+    Notes
+    -----
+    The :math:`XY`-concatenated sequence is constructed by recursively concatenating
+    control sequence structures. It's parameterized by the concatenation order :math:`l` and
+    the duration of the total sequence :math:`\tau`. Let the :math:`l`-th order of concatenation
+    be denoted as :math:`C_l(\tau)`. In this scheme, zeroth order concatenation of duration
+    :math:`\tau` is defined as free evolution over a period of :math:`\tau`. We use a notation
+    of :math:`{\mathcal 1}(\tau)` to mean free evolution over duration :math:`\tau`.
+    We define the base DSS to be:
+
+    .. math::
+        C_0(\tau) = {\mathcal 1}(\tau) \;.
+
+    The :math:`l`-th order :math:`XY`-concatenated sequence can be recursively defined as
+
+    .. math::
+        C_l(\tau) = C_{l - 1}(\tau / 4) X_{\pi} C_{l - 1}(\tau / 4) Y_{\pi}
+                    C_{l - 1}(\tau / 4) X_{\pi} C_{l - 1}(\tau / 4) Y_{\pi} \;.
```

Comment:
Interesting, how does this final `Y_{pi}` work when we have pre- and post-rotations?

Reply:
The code logic says ti would just add the X rotation as for other sequences. I am not sure whether this is reasonable in physics tho. I confess I never used this sequence before.

Reply:
Yeah, that's my reading too. I think we'll then have two pulses at the same offset though, which is a little strange. Anyway, that's not an issue for this PR.

Reply:
I suggest we leave it for now until someone has the issue with this function later.